src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -130,6 +130,9 @@ void PreReleaseRequest<I>::handle_block_writes(int r) {
     // allow clean shut down if blacklisted
     lderr(cct) << "failed to block writes because client is blacklisted"
                << dendl;
+  } else if (r == -EIO) {
```

Comment:
```-EIO``` is not bubbled up from librados.  Your commit message refers to using the admin socket to inject an ```EIO```, can you tell me how and why?

Reply:
How: I add a `LibrbdAdminSocketCommand` called `CancelOpCommand` to cancel all aio ops with the error code `-EIO`.

Why: As said in the commit message, when ceph is down and can not be put up, the librbd side (for example, the virtual machines) need to be shutdown cleanly without restarting the process. In this extreme case, we care more about the client's availability and cleanness.